File: INF4705/TP1/INF4705_H17_TP1_donnees/test_puissance.R
# Je pense néanmoins que R est plus facile à utiliser avec le package ggplot2.
# Je vous recommande très fortement d'utiliser l'IDE r-studio et de vous inspirer de ces commandes:
echo "
library(ggplot2)
library(readr)

df <- read_csv("results.csv")
df <- aggregate(df["temps"], by=df[c("algo","serie","taille")], FUN=mean)

ggplot(df, aes(x=taille, y=temps, group=algo, color=algo)) +
    geom_point() + geom_line() +
    facet_grid(. ~ serie) +
    scale_x_log10() + scale_y_log10()
ggsave("test_puissance_r.png", width=12, height=4, units="cm")
" > test_puissance.R

File: INF4705/TP1/INF4705_H17_TP1_donnees/test_puissance.py
import matplotlib.pyplot as plt
import pandas as pd
import seaborn as sns
import sys

df = pd.read_csv(sys.argv[1]).groupby(['algo','serie','taille']).mean().reset_index()

g = sns.FacetGrid(df, col='serie', hue='algo', size=4, aspect=1)
g = g.map(plt.plot, 'taille', 'temps')
g.set(xscale='log')
g.set(yscale='log')
g.add_legend()
plt.savefig('test_puissance')
